recipes/rss.tsx

```diff
@@ -228,13 +221,13 @@ const feedUpdater = handler(
     },
     required: ["items", "settings"],
   } as const satisfies JSONSchema,
-  (_event, state) => {
+  async (_event, state) => {
     if (!state.settings.feedUrl) {
       console.warn("no feed URL provided");
       return;
     }
 
-    fetchRSSFeed(
+    return await fetchRSSFeed(
```

Comment:
I don't know if we need the await here?  Isn't can't we just return the fetchRSSFeed....

(asking to learn something - I'm ok with the code as is, as it still will do the job - just perhaps with an extra async await (unless the compiler optimizes those away)

Reply:
Functionally identical to returning a promise from a non-async function, you're right -- though current lint requires an `await` for every `async`, so I've been finding myself explicitly defining a function as `async` (as an indicator a promise is returned) so things don't need to change for when another async function needs to be awaited (I don't feel strongly)

Reply:
ha, ok, same, sometimes deno/ts really wants something to be async / awaited even if it doesn't need it - to have the right signature.

Reply:
`async` function however ensures errors thrown in other parts of the function is represented as a rejected promise, rather than returning a promise, that may be resolved or rejected, *in addition to* an error being synchronously thrown

Reply:
And that is the bomb knowledge!

> **async function however ensures errors thrown in other parts of the function is represented as a rejected promise, rather than returning a promise, that may be resolved or rejected, in addition to an error being synchronously thrown**